apps/web/src/components/milestone-logs/recruitment/Milestone.tsx

```diff
@@ -11,44 +11,57 @@ import createValidator from 'class-validator-formik';
 import { Formik, Form as FormikForm } from 'formik';
 
 import { buttonBase, buttonColor, Select, Option, Field } from '@components';
-import { IENApplicantAddStatusDTO, formatDate } from '@ien/common';
-import { addMilestone, getMilestoneOptions, MilestoneType } from '@services';
-
-//@todo change any type
-const initialValues: IENApplicantAddStatusDTO = {
+import { IENApplicantAddStatusDTO, formatDate, IENStatusReasonRO } from '@ien/common';
+import {
+  addMilestone,
+  getJobAndMilestone,
+  useGetMilestoneOptions,
+  useGetWithdrawReasonOptions,
+  MilestoneType,
+} from '@services';
+
+const initialValues: any = {
   status: '',
   job_id: '',
   added_by: '',
   start_date: new Date(),
-  end_date: undefined,
   notes: '',
+  reason: '',
+  effective_date: new Date(),
 };
 
 const milestoneValidator = createValidator(IENApplicantAddStatusDTO);
 
 interface AddMilestoneProps {
   applicantId: string;
   jobId: string;
-  jobMilestones: any;
   setJobMilestones: any;
 }
 
 // Add milestone comp ***
 export const AddMilestones: React.FC<AddMilestoneProps> = ({
   applicantId,
   jobId,
-  jobMilestones,
   setJobMilestones,
 }) => {
   const handleSubmit = async (values: any, { resetForm }: any) => {
     // hardcoding some values for now, specifically logged in user
     values.job_id = jobId.toString();
     values.added_by = '1';
 
+    if (values.status !== '304' && values.status !== '305') {
```

Comment:
why should the effective_date be cleared for other statuses?

Reply:
effective_date is only show depending on those two Milestone options, otherwise it's undefined, I can't initialize a Formik Form with an undefined field, open to other solutions!